File: src/main/java/com/neutralplasma/virtusbot/storage/dataStorage/StorageHandler.java
package com.neutralplasma.virtusbot.storage.dataStorage;

import com.neutralplasma.virtusbot.storage.Info;

import java.sql.Connection;
import java.sql.SQLException;

public class StorageHandler {
    Storage storage;

    public StorageHandler(){
        if(Info.USE_MYSQL){
            storage = new MySQL();
        }else{
            storage = new SQL();
        }
        storage.openConnection();
    }

    public Connection getConnection() throws SQLException {
        return storage.getConnection();
    }

    public void createTable(String tableName, String format) throws SQLException {
        storage.createTable(tableName, format);
    }

    public void closeConnection(){
        storage.closeConnection();
    }


}

File: src/main/java/com/neutralplasma/virtusbot/storage/dataStorage/StorageHandler.kt
package com.neutralplasma.virtusbot.storage.dataStorage

import com.neutralplasma.virtusbot.storage.config.Info
import java.sql.Connection
import java.sql.SQLException
import kotlin.jvm.Throws

class StorageHandler {
    var storage: Storage? = null

    @get:Throws(SQLException::class)
    val connection: Connection?
        get() = storage?.connection

    @Throws(SQLException::class)
    fun createTable(tableName: String, format: String) {
        storage!!.createTable(tableName, format)
    }

    fun closeConnection() {
        storage!!.closeConnection()
    }

    fun execute(code: String){
        storage!!.execute(code)
    }

    init {
        storage = if (Info.USE_MYSQL) {
            MySQL()
        } else {
            SQL()
        }
        storage!!.openConnection()
    }
}